Weather.py

```diff
@@ -0,0 +1,281 @@
+
+    ''' task 01 to extract values of highest temperature and day, lowest
+    temperature and day and highest humidity and day in an year '''
+    def yearly_value_temperatures_and_humidity(self):
+        max_temp = float("-inf")
+        min_temp = float("inf")
+        max_humid = float("-inf")
+
+        for extracted_tuple in self.data:
+
+            # finding maximum temperature
+            if extracted_tuple.MaxTemperatureC != "" and max_temp < \
+                    int(extracted_tuple.MaxTemperatureC):
+                max_temp = int(extracted_tuple.MaxTemperatureC)
+                max_temp_day = self.resolve_date(extracted_tuple.PKT)
+
+            # finding minimum temperature
+            if extracted_tuple.MinTemperatureC != "" and min_temp > \
+                    int(extracted_tuple.MinTemperatureC):
+                min_temp = int(extracted_tuple.MinTemperatureC)
+                min_temp_day = self.resolve_date(extracted_tuple.PKT)
+
+            # finding maximum humidity
+            if extracted_tuple.MaxHumidity != "" and max_humid < \
+                    int(extracted_tuple.MaxHumidity):
+                max_humid = int(extracted_tuple.MaxHumidity)
+                max_humid_day = self.resolve_date(extracted_tuple.PKT)
+
+        result = \
+            [
+                max_temp, max_temp_day,
+                min_temp, min_temp_day,
+                max_humid, max_humid_day
+            ]
+
+        return result
+
+    ''' task 02 to extract values of highest average temperature, lowest
+    average temperature and average humidity in a month '''
+    def monthly_average_temperature_and_humidity(self):
+
+        high_avg_temp = 0
+        high_avg_temp_count = 0
+        low_avg_temp = 0
+        low_avg_temp_count = 0
+        mean_avg_humid = 0
+        mean_avg_humid_count = 0
+
+        for extracted_tuple in self.data:
+
+            # finding highest average temperature
+            if extracted_tuple.MaxTemperatureC != "":
+                high_avg_temp += int(extracted_tuple.MaxTemperatureC)
+                high_avg_temp_count += 1
+
+            # finding lowest average temperature
+            if extracted_tuple.MinTemperatureC != "":
+                low_avg_temp += int(extracted_tuple.MinTemperatureC)
+                low_avg_temp_count += 1
+
+            # finding mean average humidity
+            if extracted_tuple.MeanHumidity != "":
+                mean_avg_humid += int(extracted_tuple.MeanHumidity)
+                mean_avg_humid_count += 1
+
+        result = \
+            [
+                int(high_avg_temp / high_avg_temp_count),
+                int(low_avg_temp / low_avg_temp_count),
+                int(mean_avg_humid / mean_avg_humid_count)
+            ]
+
+        return result
+
+    ''' task 03 to visualize a chart for highest temperature and lowest
+    temperature of each day in a month '''
+    def monthly_chart_temperature_and_humidity(self):
```

Comment:
Look at the requirement again. You are miss-interpreting chart requirement. Do ask me if you still have confussion.